Sync/DAO/Value/ReferenceValueDAO.php

```diff
@@ -17,21 +17,41 @@ class ReferenceValueDAO
      */
     private $value;
 
+    /**
+     * @var string
+     */
+    private $type;
+
     /**
      * @return int
      */
-    public function getValue()
+    public function getValue(): ?int
     {
         return $this->value;
     }
 
     /**
      * @param int
      */
-    public function setValue($value)
+    public function setValue(int $value)
     {
         $this->value = $value;
     }
+    /**
+     * @return string
```

Comment:
```suggestion
     * @return string|null
```
if the return type is `?string` then it can be either string or null. Fix the rest of the methods and properties.